core/services/offchainreporting/OCRContract.go

```diff
@@ -0,0 +1,218 @@
+		transmitter      Transmitter
+		contractABI      abi.ABI
+	}
+
+	Transmitter interface {
+		CreateEthTransaction(ctx context.Context, toAddress gethCommon.Address, payload []byte) error
+		FromAddress() gethCommon.Address
+	}
+)
+
+var (
+	_ ocrtypes.ContractConfigTracker      = &OCRContract{}
+	_ ocrtypes.ContractTransmitter        = &OCRContract{}
+	_ ocrtypes.ContractConfigSubscription = &OCRContractConfigSubscription{}
+)
+
+func NewOCRContract(address gethCommon.Address, ethClient eth.Client, logBroadcaster eth.LogBroadcaster, jobID models.ID, transmitter Transmitter) (o *OCRContract, err error) {
+	contractFilterer, err := offchainaggregator.NewOffchainAggregatorFilterer(address, ethClient)
+	if err != nil {
+		return o, errors.Wrap(err, "could not instantiate NewOffchainAggregatorFilterer")
+	}
+
+	contractCaller, err := offchainaggregator.NewOffchainAggregatorCaller(address, ethClient)
+	if err != nil {
+		return o, errors.Wrap(err, "could not instantiate NewOffchainAggregatorCaller")
+	}
+
+	contractABI, err := abi.JSON(strings.NewReader(offchainaggregator.OffchainAggregatorABI))
+	if err != nil {
+		return o, errors.Wrap(err, "could not get contract ABI JSON")
+	}
+
+	return &OCRContract{
+		ethClient,
+		make(chan offchainaggregator.OffchainAggregatorConfigSet),
+		contractFilterer,
+		contractCaller,
+		address,
+		logBroadcaster,
+		jobID,
+		transmitter,
+		contractABI,
+	}, nil
+}
+
+func (oc *OCRContract) Transmit(ctx context.Context, report []byte, rs, ss [][32]byte, vs [32]byte) error {
+	payload, err := oc.contractABI.Pack("transmit", report, rs, ss, vs)
+	if err != nil {
+		return errors.Wrap(err, "abi.Pack failed")
+	}
+
+	return errors.Wrap(oc.transmitter.CreateEthTransaction(ctx, oc.contractAddress, payload), "failed to send Eth transaction")
+}
+
+func (oc *OCRContract) SubscribeToNewConfigs(context.Context) (ocrtypes.ContractConfigSubscription, error) {
+	sub := &OCRContractConfigSubscription{
+		make(chan ocrtypes.ContractConfig),
+		oc,
+		sync.Mutex{},
+		false,
+	}
+	connected := oc.logBroadcaster.Register(oc.contractAddress, sub)
+	if !connected {
+		return nil, errors.New("Failed to register with logBroadcaster")
+	}
+
+	return sub, nil
+}
+
+func (oc *OCRContract) LatestConfigDetails(ctx context.Context) (changedInBlock uint64, configDigest ocrtypes.ConfigDigest, err error) {
+	opts := bind.CallOpts{Context: ctx, Pending: false}
+	result, err := oc.contractCaller.LatestConfigDetails(&opts)
+	if err != nil {
+		return 0, configDigest, errors.Wrap(err, "error getting LatestConfigDetails")
+	}
+	return uint64(result.BlockNumber), ocrtypes.BytesToConfigDigest(result.ConfigDigest[:]), err
+}
+
+// Conform OCRContract to LogListener interface
+type OCRContractConfigSubscription struct {
+	ch       chan ocrtypes.ContractConfig
+	oc       *OCRContract
+	mutex    sync.Mutex
+	chClosed bool
+}
+
+func (sub *OCRContractConfigSubscription) OnConnect() {}
+func (sub *OCRContractConfigSubscription) OnDisconnect() {
```

Comment:
Would it be worth logging these? Are there any other places where we should be emitting logs?

Reply:
Good question for @spooktheducks

Reply:
LogBroadcaster logs them, so you'd just end up with dupes